mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp

```diff
@@ -215,11 +215,14 @@ spirv::Deserializer::processMemoryModel(ArrayRef<uint32_t> operands) {
   if (operands.size() != 2)
     return emitError(unknownLoc, "OpMemoryModel must have two operands");
 
+  StringRef addressing_model = module->getAddressingModelAttrName().strref();
```

Comment:
Why StringRef instead of StringAttr?

Reply:
should I alter this all across the place?

Reply:
Yes

Reply:
done